src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -1,138 +1,91 @@
+      storeActor ! CallCompleted(completedCall, callOutputs)
+      stay()
+    case Event(StoreActor.RunnableCalls(runnableCalls), NoFailureMessage) =>
       if (runnableCalls.nonEmpty) {
         log.info("Starting calls: " + runnableCalls.map {_.name}.toSeq.sorted.mkString(", "))
       }
       runnableCalls foreach startCallActor
-    }
+      stay()
+    case Event(CallActor.Failed(call, failure), NoFailureMessage) =>
+      storeActor ! UpdateStatus(call, ExecutionStatus.Failed)
+      goto(WorkflowFailed) using FailureMessage(failure)
+    case Event(WorkflowActor.Complete, NoFailureMessage) => goto(WorkflowSucceeded)
   }
 
-  /**
-   * Performs the following steps:
-   * <ol>
-   *   <li>Puts this actor into its "started" state.</li>
-   *   <li>Starts all runnable calls.</li>
-   *   <li>Messages the sender that the workflow has been started.</li>
-   * <ol>
-   */
-  private def handleInitialStart(): Unit = {
-    context.become(started)
-    startRunnableCalls()
-    sender ! WorkflowActor.Started
+  when(WorkflowFailed) {
+    case Event(GetFailureMessage, msg: FailureMessage) =>
+      sender() ! msg
+      stay()
   }
 
-  /**
-   * Coordinate handling of the completion of this call with the store actor.  The invocation of this
-   * method should result in the outputs of the call being written to the symbol store and the status
-   * of the call being marked complete.  If all calls in the workflow are complete the parent of this
-   * actor will be messaged a `WorkflowActor.Done` with the workflow outputs.
-   */
-  private def handleCallCompleted(completedCall: Call, callOutputs: WorkflowOutputs): Unit = {
-    // Messages the store actor with a `CallCompleted` via an `ask`; i.e. this is not *asking* the
-    // store actor if the call is completed, it's informing the store actor of that fact via `ask`.
-    // The store actor responds with a `Future[Boolean]` which if `true` means the workflow has completed.
-    (storeActor ? CallCompleted(completedCall, callOutputs)) onComplete {
-      case Success(done) =>
-        done match {
-          case true =>
-            val futureDoneMessage = for {
-              outputs <- (storeActor ? GetOutputs).mapTo[Map[String, WdlValue]]
-            } yield WorkflowActor.Done(outputs)
-            futureDoneMessage pipeTo context.parent
+  // We're supporting GetOutputs for all states, so there's nothing particular to do here
+  when(WorkflowSucceeded)(FSM.NullFunction)
 
-          case _ => startRunnableCalls()
-        }
-      case Failure(t) =>
-        log.error(t, t.getMessage)
-        context.parent ! WorkflowFailed
-    }
+  whenUnhandled {
+    case Event(WorkflowActor.GetOutputs, _) =>
+      storeActor.ask(StoreActor.GetOutputs) pipeTo sender
+      stay()
+    case Event(e, _) =>
+      log.debug(s"Received unhandled event $e while in state $stateName")
+      stay()
+  }
+
+  onTransition {
+    case WorkflowSubmitted -> WorkflowRunning => storeActor ! FindRunnableCalls
```

Comment:
Yeah. You do need to seed the process somehow this way. I considered renaming it to reflect that it's more of a get-the-call-actor-ball-rolling affair but *shrug*